tests/ci/cdk/cdk/windows_docker_image_build_stack.py

```diff
@@ -1,66 +1,124 @@
-        inline_policies = {"ecr_power_user_policy": ecr_power_user_policy, "s3_read_write_policy": s3_read_write_policy}
-        role = iam.Role(scope=self, id="{}-role".format(id),
-                        assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"),
-                        inline_policies=inline_policies,
-                        managed_policies=[
-                            iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSSMManagedInstanceCore")
-                        ])
+        ecr_power_user_policy = iam.PolicyDocument.from_json(
+            ecr_power_user_policy_in_json([WINDOWS_X86_ECR_REPO])
+        )
+        s3_read_write_policy = iam.PolicyDocument.from_json(
+            s3_read_write_policy_in_json(S3_BUCKET_NAME)
+        )
+        inline_policies = {
+            "ecr_power_user_policy": ecr_power_user_policy,
+            "s3_read_write_policy": s3_read_write_policy,
+        }
+        role = iam.Role(
+            scope=self,
+            id="{}-role".format(id),
+            assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"),
+            inline_policies=inline_policies,
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonSSMManagedInstanceCore"
+                )
+            ],
+        )
 
         # Define Windows EC2 instance, where the SSM document will be executed.
-        # TODO: This AMI does not have docker installed by default anymore. Find another Windows machine
-        #       that has docker by default or update the ssm document to properly install docker.
         machine_image = ec2.MachineImage.latest_windows(
-            ec2.WindowsVersion.WINDOWS_SERVER_2019_ENGLISH_FULL_BASE)
+            ec2.WindowsVersion.WINDOWS_SERVER_2019_ENGLISH_FULL_BASE
+        )
         vpc = ec2.Vpc(scope=self, id="{}-vpc".format(id))
-        block_device_volume = ec2.BlockDeviceVolume.ebs(volume_size=200, delete_on_termination=True)
-        block_device = ec2.BlockDevice(device_name="/dev/sda1", volume=block_device_volume)
-        instance = ec2.Instance(scope=self,
-                                id="{}-instance".format(id),
-                                instance_type=ec2.InstanceType(instance_type_identifier="m5d.xlarge"),
-                                vpc=vpc,
-                                role=role,
-                                block_devices=[block_device],
-                                vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PUBLIC),
-                                machine_image=machine_image)
+        block_device_volume = ec2.BlockDeviceVolume.ebs(
+            volume_size=200, delete_on_termination=True
+        )
+        block_device = ec2.BlockDevice(
+            device_name="/dev/sda1", volume=block_device_volume
+        )
+
+        setup_user_data = ec2.UserData.for_windows()
```

Comment:
Are there any gotchas or things to know about "user data"? This seems like something really useful I could steal for the ec2 test framework.